part3/identifying-high-leverage-activities.qmd

```diff
@@ -0,0 +1,284 @@
+| **Product Agent** | Product Information | Manages queries about plans, devices, and promotions |
+
+Additionally, we should also map out the tools we can provide to the agent
+
+**Billing sub-agent setup**
+
+![Billing agent tools](./files/billing-agent-tools.png)
+
+### 3. Identify available data sources 
+
+Structured data sources
+
+| Data Source | Schema Summary |
+|-------------|----------------|
+| telco_customer_support_dev.bronze.customers | Customer profile data including customer_id, segment, location, registration_date, status, contact preferences, and scoring metrics (loyalty_tier, churn_risk_score, customer_value_score, satisfaction_score) |
+| telco_customer_support_dev.bronze.subscriptions | Subscription records linking customers to plans/devices with subscription_id, customer_id, plan_id, device_id, promo_id, dates, contract terms, charges, and status |
+| telco_customer_support_dev.bronze.plans | Service plan catalog containing plan_id, name, type, pricing, data limits, feature flags (unlimited calls/texts), contract requirements, and descriptions |
+| telco_customer_support_dev.bronze.devices | Device catalog with device_id, name, manufacturer, type, pricing (retail/installment), specifications (storage, 5G compatibility), colors, and availability status |
+| telco_customer_support_dev.bronze.promotions | Promotional offers including promo_id, name, discount details (type/value), date ranges, descriptions, and active status |
+| telco_customer_support_dev.bronze.billing | Billing records with billing_id, customer/subscription references, dates, charge breakdowns (base, additional, tax), payment information, and status |
+| telco_customer_support_dev.bronze.usage | Usage metrics containing usage_id, subscription_id, daily usage data (data_usage_mb, voice_minutes, sms_count), and billing cycle information |
+
+
+Unstructured data sources 
+
+| Table | Unstructured Content |
+----------|-------------------------------|
+| telco_customer_support_dev.bronze.knowledge_base | Full content stored as markdown text including FAQs, policies, guides, and procedures |
+| telco_customer_support_dev.bronze.support_tickets | Free-text issue descriptions from customers |
+| telco_customer_support_dev.bronze.support_tickets | Free-text resolution details from support agents |
+
+
+### Instrumentating observability based on personas
+
+TODO: something here about involving stakeholders early in the planning process and providing good information with user interfaces for internal stakeholders? 
+
+In the chapter Principle 3: The technical practices of learning, we talked at length about how to use tracing to provide visibility into the workings of our agentic application. In this section, we want to expand on this section to cover how traces might be used to provide different stakeholders with a view on how the agent is functioning, in a way that addresses what they care about. 
```

Comment:
It would be great to specifically discuss Tracing and User Feedback Collection as the two core solutions for gathering the data required by stakeholders. It would also be very helpful to discuss Databricks-specific offerings for this: (1) MLflow Tracing and (2) Assessment / Feedback APIs.